File: truss-compiler/src/main/java/truss/compiler/symbols/InvalidTypeSymbol.java
package truss.compiler.symbols;

public class InvalidTypeSymbol extends TypeSymbolBase {
    public static final String INVALID_NAME = "**INVALID TYPE**";

    public InvalidTypeSymbol(ContainerSymbol parent) {
        super(INVALID_NAME, INVALID_NAME, parent);
    }

    @Override
    public TypeKind getTypeKind() {
        return TypeKind.INVALID;
    }

    @Override
    public SymbolKind getKind() {
        return SymbolKind.INVALID_TYPE;
    }

    @Override
    public void accept(SymbolVisitor visitor) {
        if (!visitor.isDone()) {
            visitor.visitInvalidType(this);
        }
    }
}

File: Truss.Compiler/Symbols/InvalidTypeSymbol.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Truss.Compiler.Symbols {
    public class InvalidTypeSymbol : TypeSymbol {
        public const string InvalidName = "**INVALID TYPE**";

        public InvalidTypeSymbol(ContainerSymbol parent)
            : base(InvalidName, InvalidName, parent) {
        }

        public override TypeKind TypeKind {
            get { return TypeKind.Invalid; }
        }

        public override void Accept(ISymbolVisitor visitor) {
            if (!visitor.Done) {
                visitor.VisitInvalidType(this);
            }
        }
    }
}
